qiita_db/metadata_template.py

```diff
@@ -1445,6 +1462,89 @@ def extend(self, md_template):
         # adding the fp to the object
         self.add_filepath(fp)
 
+    def update(self, md_template):
+        r"""Update values in the sample metadata template
+
+        Parameters
+        ----------
+        md_template : DataFrame
+            The metadata template file contents indexed by samples Ids
+
+        Raises
+        ------
+        QiitaDBError
+            If md_template and db do not have the same sample ids
+            If md_template and db do not have the same column headers
+        """
+        conn_handler = SQLConnectionHandler()
+
+        # Clean and validate the metadata template given
+        new_map = self._clean_validate_template(md_template, self.id,
+                                                conn_handler)
+        # Retrieving current metadata
+        current_map = self._transform_to_dict(conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.{0} WHERE {1}=%s".format(self._table,
+                                                          self._id_column),
+            (self.id,)))
+        dyn_vals = self._transform_to_dict(conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.{0}".format(self._table_name(self.id))))
+
+        for k in current_map:
+            current_map[k].update(dyn_vals[k])
+            try:
+                del current_map[k]['study_id']
+            except KeyError:
+                pass
+
+        # converting sql results to dataframe
+        current_map = pd.DataFrame.from_dict(current_map, orient='index')
+
+        # simple validations of sample ids and column names
+        samples_diff = set(
+            new_map.index.tolist()) - set(current_map.index.tolist())
+        if samples_diff:
+            raise QiitaDBError('The new sample template differs from what is '
+                               'stored in database by these samples names: %s'
+                               % ', '.join(samples_diff))
+        columns_diff = set(new_map.columns) - set(current_map.columns)
+        if columns_diff:
+            raise QiitaDBError('The new sample template differs from what is '
+                               'stored in database by these columns names: %s'
+                               % ', '.join(columns_diff))
+
+        # here we are comparing two dataframes following:
+        # http://stackoverflow.com/questions/24495695
```

Comment:
This is pointing to a question, would you mind pointing to the answer that you are referring to?